test/system/src/test/java/io/pravega/test/system/MultiReaderWriterWithFailOverTest.java

```diff
@@ -0,0 +1,307 @@
+        log.debug("Zookeeper service details: {}", zkUris);
+        //get the zk ip details and pass it to bk, host, controller
+        URI zkUri = zkUris.get(0);
+
+        //2. Start 3 bookies
+        Service bkService = new BookkeeperService("bookkeeper", zkUri);
+        if (!bkService.isRunning()) {
+            bkService.start(true);
+        }
+        List<URI> bkUris = bkService.getServiceDetails();
+        log.debug("Bookkeeper service details: {}", bkUris);
+
+        //3. start 2 instances of pravega controller
+        Service conService = new PravegaControllerService("controller", zkUri);
+        if (!conService.isRunning()) {
+            conService.start(true);
+        }
+        conService.scaleService(2, true);
+        List<URI> conUris = conService.getServiceDetails();
+        log.debug("Pravega Controller service  details: {}", conUris);
+
+        //4.start 2 instances of pravega segmentstore
+        Service segService = new PravegaSegmentStoreService("segmentstore", zkUri, conUris.get(0));
+        if (!segService.isRunning()) {
+            segService.start(true);
+        }
+        segService.scaleService(2, true);
+        List<URI> segUris = segService.getServiceDetails();
+        log.debug("Pravega segmentstore service  details: {}", segUris);
+    }
+
+    @Before
+    public void setup() {
+
+        //1. Start 1 instance of zookeeper
+        Service zkService = new ZookeeperService("zookeeper");
+        if (!zkService.isRunning()) {
+            zkService.start(true);
+        }
+        List<URI> zkUris = zkService.getServiceDetails();
+        log.debug("Zookeeper service details: {}", zkUris);
+        //get the zk ip details and pass it to bk, host, controller
+        URI zkUri = zkUris.get(0);
+
+        // Verify controller is running.
+        controllerInstance = new PravegaControllerService("controller", zkUri);
+        assertTrue(controllerInstance.isRunning());
+        List<URI> conURIs = controllerInstance.getServiceDetails();
+        log.info("Pravega Controller service instance details: {}", conURIs);
+
+        // Verify segment store is running.
+        segmentStoreInstance = new PravegaSegmentStoreService("segmentstore", zkUri, conURIs.get(0));
+        assertTrue(segmentStoreInstance.isRunning());
+        log.info("Pravega segment store instance details: {}", segmentStoreInstance.getServiceDetails());
+    }
+
+    @After
+    public void tearDown() {
+        controllerInstance.stop();
+        segmentStoreInstance.stop();
+    }
+
+    @Test(timeout = 600000)
+    public void multiReaderWriterWithFailOverTest() throws Exception {
+
+        log.info("Test with 2 controller, SSS instances running and without a failover scenario");
+        readWriteTest();
+
+        //scale down SSS by 1 instance
+        segmentStoreInstance.scaleService(1, true);
+        Thread.sleep(60000);
+        log.info("Test with 1 SSS instance down");
+        readWriteTest();
```

Comment:
Answering to the above comment:
The same method readWriteTest() is called multiple times after different failovers.
L144-> test with 2SSS, 2 controller instances (2S,2C)
L150-> test with 1SSS failover(2S,2C)->(1S,2C)
L158-> test with 1 controller failover(1S,2C)->(2S,2C)->(2S,1C)
L164-> test with 1 controller, 1SSS instance down (2S,1C)->(1S,1C).
*(A,B) -> A- #instances of pravega Segmentstore service , B- #instances of pravega controller service.